src/clj/puppetlabs/services/jruby_pool_manager/impl/reference_pool.clj

```diff
@@ -131,7 +133,7 @@
           (jruby-agents/send-agent modify-instance-agent
                                    #(flush-if-at-max-borrows pool-context instance)))
         ;; Return the worker-id, to be used in metrics and event logging
-        (.getId (Thread/currentThread)))))
+        (pool-protocol/worker-id pool-context instance))))
```

Comment:
I think we might not need to do this anymore, if you're calling `worker-id` explicitly in the return event registering?

Reply:
This is helpful in testing that the worker-id is the same at borrow-time and return-time.

Though the real reason I kept it was to change as little as possible.. 😁

Reply:
Okay I think it's fine for now then.